x-pack/plugins/code/server/repository_service.ts

```diff
@@ -261,29 +266,37 @@ export class RepositoryService {
           return 0;
         };
       }
-      const gitRepo = await Git.Clone.clone(repo.url, localPath, {
-        bare: 1,
-        fetchOpts: {
-          callbacks: cbs,
-        },
-      });
-      const headCommit = await gitRepo.getHeadCommit();
-      const headRevision = headCommit.sha();
-      const currentBranch = await gitRepo.getCurrentBranch();
-      const currentBranchName = currentBranch.shorthand();
-      this.log.info(
-        `Clone repository from ${
-          repo.url
-        } done with head revision ${headRevision} and default branch ${currentBranchName}`
-      );
-      return {
-        uri: repo.uri,
-        repo: {
-          ...repo,
-          defaultBranch: currentBranchName,
-          revision: headRevision,
-        },
-      };
+      let gitRepo: Git.Repository | undefined;
+
+      try {
+        gitRepo = await Git.Clone.clone(repo.url, localPath, {
+          bare: 1,
+          fetchOpts: {
+            callbacks: cbs,
+          },
+        });
+        const headCommit = await gitRepo.getHeadCommit();
+        const headRevision = headCommit.sha();
+        const currentBranch = await gitRepo.getCurrentBranch();
+        const currentBranchName = currentBranch.shorthand();
+        this.log.info(
+          `Clone repository from ${
+            repo.url
+          } done with head revision ${headRevision} and default branch ${currentBranchName}`
+        );
+        return {
+          uri: repo.uri,
+          repo: {
+            ...repo,
+            defaultBranch: currentBranchName,
+            revision: headRevision,
+          },
+        };
+      } finally {
+        if (gitRepo) {
+          gitRepo.cleanup();
```

Comment:
no need?

Reply:
for the git repo created by git clone/update, let's leave them out of the repo object cache now. since they are not a part of `GitOperations` and they are not used very often in the lifecycle of a repo.